### PR TITLE
Fixes buckling issue on borgs

### LINKED ISSUE #52404: Ghosts can buckle people to cyborgs
Issue reported from  Round ID: [142464](https://scrubby.melonmesa.com/round/142464) (/tg/Station Basil [ENGLISH] [US-WEST] [100% FREE LAG])

Reporting client version: 513.1526
On shift 142457 Me and other people could buckle people to cyborgs as ghosts. I didn't get to test it but I could do it to a dead guy buckled into a stasis, onto a mediborg. I observed but a guy who died naturally could do it too. I'd include more images and stuff but it was in a previous shift. This could probably get abused pretty hard.

### PR DESCRIPTION
**Pull request by nightred**
## About The Pull Request

You must be living to buckle to a borg.

Fixes #52404

## Why It's Good For The Game

Fixes bugs.

## Changelog
:cl:
fix: Only living mobs can buckle to borgs
/:cl:

### CODE REVIEW

**Comment by TheWiznard** [comment]
does this change borgs buckling a body to them on their own if so rip

**Comment by nightred** [comment]
> does this change borgs buckling a body to them on their own if so rip

Dead bodies are still part of the /mob/living/ group, this still works as expected. this only validates that the person doing the dragging is a living mob.

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 1]
```diff
@@ -995,7 +995,7 @@
 	cell = null
 
 /mob/living/silicon/robot/mouse_buckle_handling(mob/living/M, mob/living/user)
-	if(can_buckle && istype(M) && !(M in buckled_mobs) && ((user!=src)||(a_intent != INTENT_HARM)))
+	if(can_buckle && istype(user) && istype(M) && !(M in buckled_mobs) && ((user!=src)||(a_intent != INTENT_HARM)))
```
Comment (kriskog) [inline code/modules/mob/living/silicon/robot/robot.dm:998]:
```suggestion
	if(can_buckle && isliving(user) && isliving(M) && !(M in buckled_mobs) && ((user!=src)||(a_intent != INTENT_HARM)))
```
Same thing, but slightly more readable

Comment (nightred) [reply code/modules/mob/living/silicon/robot/robot.dm]: I agree it does look nicer.

**Review by kriskog — APPROVED** [review]

**Pull request merged**